lib/matplotlib/axes/_axes.py

```diff
@@ -6091,25 +6091,35 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
                              "Please only use 'density', since 'normed'"
                              "will be deprecated.")
 
-        # process the unit information
-        self._process_unit_info(xdata=x, kwargs=kwargs)
-        x = self.convert_xunits(x)
-        if bin_range is not None:
-            bin_range = self.convert_xunits(bin_range)
-
-        # Check whether bins or range are given explicitly.
-        binsgiven = (cbook.iterable(bins) or bin_range is not None)
-
         # basic input validation
         input_empty = np.size(x) == 0
-
         # Massage 'x' for processing.
         if input_empty:
             x = np.array([[]])
         else:
             x = cbook._reshape_2D(x, 'x')
         nx = len(x)  # number of datasets
 
+        # Process unit information
+        # If doing a stacked histogram, the input is a list of datasets, so
+        # we need to do the unit conversion individually on eaach dataset
+        if stacked:
+            self._process_unit_info(xdata=x[0], kwargs=kwargs)
+            newx = []
+            for xi in x:
+                newx.append(self.convert_xunits(xi))
```

Comment:
`x = [self.convert_xunits(xi) for xi in x]` ?

Reply:
🤦‍♂️ thanks...